### PR TITLE
[feat] engine: implementation of searchmysite

### PR DESCRIPTION
**Pull request by Austin-Olacsi**
## What does this PR do?

Adds searchmysite engine

## Why is this change important?

it was requested on the old project searx: https://github.com/searx/searx/issues/2817
the content is mostly personal blogs that don't show up in the results of other search engines. 

## How to test this PR locally?

make run
!sms android

### CODE REVIEW

**Review by return42 — APPROVED** [review]
thanks :+1:

File: searx/settings.yml [hunk 1]
```diff
@@ -1563,6 +1563,19 @@ engines:
...
   #    disabled: true
 
+  - name: searchmysite
+    engine: xpath
+    shortcut: sms
+    categories: general
+    paging: true
+    search_url: https://searchmysite.net/search/?q={query}&page={pageno}
+    url_xpath: //p[@class='sms-p3']/a[@class='result-link']/text()
+    title_xpath: //span[@class='result-title-txt']/text()
```
Comment (Bnyro) [inline searx/settings.yml]:
```suggestion
    title_xpath: //span[contains(@class,'result-title-txt')]/text()
```
makes it more robust against changes at their side

I would do the same for the classes in the url and content xpath

**Comment by Bnyro** [comment]
I think that it would be more robust to use a `results_xpath` (you'll find enough other engines in SearXNG for reference) - it just lowers chances of getting mixed results (like the url from one result but the title of another)

File: searx/settings.yml [hunk 2]
```diff
@@ -1563,6 +1563,19 @@ engines:
...
+    content_xpath: //*[@id='result-hightlight']
```
Comment (Bnyro) [inline searx/settings.yml]: Why do you use a wildcard selector here?

**Comment by Bnyro** [comment]
> I think that it would be more robust to use a `results_xpath` (you'll find enough other engines in SearXNG for reference) - it just lowers chances of getting mixed results (like the url from one result but the title of another)
>
@Austin-Olacsi I've quickly pushed my suggestions, you can find these changes here: https://github.com/searxng/searxng/compare/07bdea1d95d8c87c9c4959c4a83ab4febb54a5a6..d7693e65d271c9a4f1b8c64b0f0e4df9faa55952.

**Review by Bnyro — APPROVED** [review]
Looks good to me, thanks for the PR :+1:

**Pull request merged**